app/models/merchant.rb

```diff
@@ -16,10 +16,19 @@ def disabled_items
   end
 
   def self.enabled_merchants
-    all.where(status: "enabled")
+    where(status: "enabled")
   end
 
   def self.disabled_merchants
-    all.where(status: "disabled")
+    where(status: "disabled")
+  end
+
+  def self.top_five_by_revenue
+    joins(:items, invoices: :transactions)
+      .select('merchants.*, sum(invoice_items.unit_price * invoice_items.quantity) as revenue')
```

Comment:
You said you weren't sure about this part:  Revenue for an invoice should be calculated as the sum of the revenue of all invoice items

But I do beleive that your aggregate in your select statement does this.